basxconnect/core/views/person/person_modals_views.py

```diff
@@ -156,13 +157,32 @@ def get_layout(self):
 class EditPostalAddressView(EditView):
     fields = ["type", "address", "postcode", "city", "country"]
 
-    def get_success_url(self):
-        return reverse_model(
-            self.object.person, "read", kwargs={"pk": self.object.person.pk}
+    def get_form(self, form_class=None):
+        ret = super().get_form(form_class)
+        ret.fields["is_primary"] = django.forms.BooleanField(
+            label=_("Use as primary postal address"), required=False
         )
+        return ret
+
+    def form_valid(self, *args, **kwargs):
+        ret = super().form_valid(*args, **kwargs)
+        is_primary = "is_primary" in args[0].changed_data
```

Comment:
Nitpick: I would specify the ```form``` parameter arg[0] explicitly here, it is easier to understand what is going on.

Reply:
Okay. Sorry, I overread the ```changed_data```. I know now why I used ```get_form_class``` in the past: Because only by overriding the class, all fields will correctly be cleaned. We have to override ```get_form_class```. You can take a look at how I did it with the password dialog: 
https://github.com/basxsoftwareassociation/bread/blob/main/bread/views/userprofile.py#L174

Reply:
You can either define an inline class or use ```type()``` to create a new subclass.